runtime/common/src/lib.rs

```diff
@@ -395,6 +395,23 @@ where
 	}
 }
 
+pub struct EvmLimits<T>(PhantomData<T>);
+impl<T> EvmLimits<T>
+where
+	T: frame_system::Config,
+{
+	pub fn max_gas_limit() -> u64 {
+		let weights = T::BlockWeights::get();
+		let normal_weight = weights.get(DispatchClass::Normal);
+		WeightToGas::convert(normal_weight.max_extrinsic.unwrap_or(weights.max_block))
+	}
+
+	pub fn max_storage_limit() -> u32 {
+		let length = T::BlockLength::get();
+		*length.max.get(DispatchClass::Normal)
+	}
+}
+
```

Comment:
Can you add integration test to get the `max_gas_limit` and `max_storage_limit` value?